src/contexts/providers.context.tsx

```diff
@@ -18,38 +18,55 @@ import {
   isMetaMaskResourceUnavailableError,
   isMetaMaskUnknownChainError,
   isMetaMaskUserRejectedRequestError,
+  isAsyncTaskDataAvailable,
 } from "src/utils/types";
-import { ethereumAccountsParser, getConnectedAccounts } from "src/adapters/ethereum";
+import {
+  ethereumAccountsParser,
+  getConnectedAccounts,
+  silentlyGetConnectedAccounts,
+} from "src/adapters/ethereum";
 import { useEnvContext } from "src/contexts/env.context";
 import { useErrorContext } from "src/contexts/error.context";
 import routes from "src/routes";
-import { Chain, EthereumChainId, EthereumEvent, WalletName } from "src/domain";
+import { Env, Chain, EthereumChainId, EthereumEvent, WalletName } from "src/domain";
 
 interface ProvidersContext {
-  connectedProvider?: { provider: Web3Provider; chainId: number };
+  connectedProvider: AsyncTask<
+    {
+      provider: Web3Provider;
+      chainId: number;
```

Comment:
Probably it would make sense to add the `account` here as well instead of having it separately, as the `connectedProvider` that we are using is strictly connected to an `account`.

Reply:
I would create a separate interface for this in our domain btw.

Reply:
Although it makes sense, this is a significant change. Are you sure about going it in this PR?

I can intuit changes in the behavior of certain reloads/renders in useEffects derived from the change of identity of the object in their dependencies.

Reply:
If those changes happens in the `useEffect` then it's perfect! That's how they should work, if we change the `connectedProvider` and hence the `account` data, things depending on the `account` should re-render.

I would try to do it in this PR as the last thing, and if we see that things get complicated I'll move it to another PR.

Reply:
Since this task implies several changes, I opened PR #212 pointing to this one so you can review these changes in isolation.